### DISCUSSION TITLE
How to use usePagination hook ?

### DISCUSSION THREAD

**Discussion by Sakkhor909** [mantinedev/mantine#810] [Q&A] [answered]
I have a Pagination component from `import { Pagination } from "@mantine/core";`
 How to Do I control its state by this hook `import { usePagination } from '@mantine/hooks';`
I have read this but got no idea how these two are related https://mantine.dev/hooks/use-pagination/

**Comment by rtivital** [comment, marked as answer]
You do not need usePagination hook to control Pagination component state. Pagination component is built on top of usePagination hook. See controlled example to learn how to get Pagination state – https://mantine.dev/core/pagination/#controlled

**Reply by rashdev10** [reply]
It is very challenging with the documentation to figure out how I can use the pagination. There is no example code to show how the pagination really works. I was lost on how I am going to pass the state i.e data I want to paginate. How do I pass the data, I have no clue. 
@rtivital

**Reply by rtivital** [reply]
https://codesandbox.io/s/beautiful-star-o0wcx?file=/src/App.tsx

**Reply by Danztee** [reply]
the pagination is not working... even with the example

**Reply by Danztee** [reply]
The example is not working now... event he mantine version used there is version 3 and we're in 6 and you expect it to work?

**Reply by rtivital** [reply]
Version 6 works fine – https://codesandbox.io/s/pensive-wing-jw9xej?file=/src/App.tsx
